### PR TITLE
feat(chainguard): add support for Chainguard distro

### PR DESCRIPTION
**Pull request by luhring**
Adds support for incorporating Chainguard secdb data into Trivy DB for https://github.com/aquasecurity/trivy/issues/3630

### CODE REVIEW

File: pkg/vulnsrc/chainguard/chainguard.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+func (vs VulnSrc) saveSecFixes(tx *bolt.Tx, platform, pkgName string, secfixes map[string][]string) error {
+	for fixedVersion, vulnIDs := range secfixes {
+		advisory := types.Advisory{
+			FixedVersion: fixedVersion,
+		}
+		for _, vulnID := range vulnIDs {
+			// See https://gitlab.alpinelinux.org/alpine/infra/docker/secdb/-/issues/3
+			// e.g. CVE-2017-2616 (+ regression fix)
```
Comment (knqyf263) [inline pkg/vulnsrc/chainguard/chainguard.go:93]: question: Is it relevant for Chainguard?

File: pkg/vulnsrc/chainguard/chainguard.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+const (
+	chainguardDir = "chainguard"
+	distroName    = "chainguard"
+)
+
+var (
+	source = types.DataSource{
+		ID:   vulnerability.Chainguard,
+		Name: "Chainguard Secdb",
```
Comment (knqyf263) [inline pkg/vulnsrc/chainguard/chainguard.go:26]: nit: It is called "Chainguard Security Data" here. Which is the correct name for Chainguard security advisories?

File: pkg/vulnsrc/chainguard/chainguard.go [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	for fixedVersion, vulnIDs := range secfixes {
+		advisory := types.Advisory{
+			FixedVersion: fixedVersion,
+		}
+		for _, vulnID := range vulnIDs {
+			// See https://gitlab.alpinelinux.org/alpine/infra/docker/secdb/-/issues/3
+			// e.g. CVE-2017-2616 (+ regression fix)
+			ids := strings.Fields(vulnID)
+			for _, cveID := range ids {
+				cveID = strings.ReplaceAll(cveID, "CVE_", "CVE-")
```
Comment (knqyf263) [inline pkg/vulnsrc/chainguard/chainguard.go:96]: question: Also, it is for sanitizing Alpine secdb. Do you think it is needed in Chainguard?

Comment (luhring) [reply pkg/vulnsrc/chainguard/chainguard.go:96]: That's a good point. We should be able to guarantee that we're not putting things like `CVE_` in our data. I can remove this 👍

Comment (luhring) [reply pkg/vulnsrc/chainguard/chainguard.go]: Pushed a change to simplify this — let me know what you think

File: pkg/vulnsrc/chainguard/chainguard.go (hunk 1 above)
Comment (luhring) [reply pkg/vulnsrc/chainguard/chainguard.go]: I don't think it has to be — pushed a change that removes this.

File: pkg/vulnsrc/chainguard/chainguard.go (hunk 2 above)
Comment (knqyf263) [reply pkg/vulnsrc/chainguard/chainguard.go:26]: Sorry I forgot to put a link. We want to be consistent in what we call it across our projects.
https://github.com/aquasecurity/vuln-list-update/blob/c98ad9e9909ecb97de0f8777131c251023167c81/main.go#L245

Comment (luhring) [reply pkg/vulnsrc/chainguard/chainguard.go]: Pushed an update!

**Comment by knqyf263** [comment]
As I commented [here](https://github.com/aquasecurity/vuln-list-update/pull/190#issuecomment-1483065015), we are planning to cut v0.40.0 shortly.

**Comment by knqyf263** [comment]
@luhring What does `0` mean? It will never be detected. Is it intended?
https://github.com/aquasecurity/vuln-list/blob/842558b6dcc7cfa5934bdf353212b212e6926b36/chainguard/chainguard/git.json#L4-L6

**Comment by luhring** [comment]
> @luhring What does `0` mean? It will never be detected. Is it intended?

Yes, we put `0` in the secdb to NAK vulnerabilities that scanners find, to let scanners know that no versions of the given package should be matched to the vulnerabilities. This is the same behavior as Wolfi and as Alpine.

**Comment by knqyf263** [comment]
:shipit:

**Pull request merged**